### PR TITLE
docs: add upgrade guide for go 1.x-rc.x

### PR DESCRIPTION
**Pull request by JenniferMah**
Jira ticket: https://issues.corp.twilio.com/browse/DI-2037

### CODE REVIEW

**Review by beebzz — COMMENTED** [review]
Looks good, but didn't we want to use `X.0.0-rc` as the rc branch naming convention?

**Comment by JenniferMah** [comment]
Yup! I just wanted to verify what our first version number will be before I renamed the branch since I'll have to merge over the exiting work on release-candidate to the new x.0.0-rc branch anyways!

**Review by beebzz — APPROVED** [review]

File: UPGRADE.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+params := &api.CreateMessageParams{}
+params.SetFrom("+15017122661")
+params.SetBody("body")
+params.SetTo("+15558675310")
+
+resp, err := client.ApiV2010.CreateMessage(params)
+```
+```go
+//1.x.xrc
```
Comment (rakatyal) [inline UPGRADE.md:23]: Think we should write this as 1.0.0-rc.1 since this will get released in our 1st RC build.

Comment (JenniferMah) [reply UPGRADE.md:23]: I can update it to `1.x.x-rc.x` but I think by changing this to `1.0.0-rc.1` some users could assume it's only a change that was applied to that specific version. We also use the same naming conventions in our [other upgrade guides](https://github.com/twilio/twilio-java/blob/main/UPGRADE.md#2020-09-28-7xx-to-8xx) to help make it clear that this  will apply for all versions higher than 0.2.x.

Comment (rakatyal) [reply UPGRADE.md:23]: Okay that makes sense, and we should keep it consistent with other guides. So feel free to disregard the suggestion.

**Review by rakatyal — APPROVED** [review]

File: UPGRADE.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Upgrade Guide
+
+_All `MAJOR` version bumps will have upgrade notes posted here._
+
+[2022-04-27] 0.2.x to 1.x.xrc
```
Comment (childish-sambino) [inline UPGRADE.md:5]: This needs to be updated to the date when the actual RC version is released for librarian to be able to find it.

**Pull request merged**